clientconn.go

```diff
@@ -123,10 +126,15 @@ func WithInitialConnWindowSize(s int32) DialOption {
 	}
 }
 
-// WithMaxMsgSize returns a DialOption which sets the maximum message size the client can receive.
+// WithMaxMsgSize returns a DialOption which sets the maximum message size the client can receive. Deprecated: use WithMaxReceiveMessageSize instead.
```

Comment:
`WithMaxReceiveMessageSize` is not a DialOption anymore.
We could say, "use WithDefaultCallOptions(WithMaxReceiveMessageSize(s)) instead",
or "use CallOption WithDefaultCallOptions instead".

Also, see my following comment, `WithDefaultCallOptions` should be renamed.